test/watcher_test.js

```diff
@@ -3,8 +3,28 @@ import chai from 'chai';
 import sinonChai from 'sinon-chai';
 import Sinon from 'sinon';
 import Watcher from '../lib/watcher';
+import broccoli from '..';
+const Builder = broccoli.Builder;
+const multidepRequire = require('multidep')('test/multidep.json');
 import SourceNodeWrapper from '../lib/wrappers/source-node';
 
+const Plugin = multidepRequire('broccoli-plugin', '1.3.0');
+const Merge = multidepRequire('broccoli-merge-trees', '4.1.0');
```

Comment:
Why are we requiring these specific versions? Is it that these changes only work with those versions? If yes, then the change may be incompatible generally... 🤔 

We should probably be running these tests within a `multidepRequire.forEachVersion` loop...

Reply:
It is simply to create dummy plugins and merge them into one output. There is probably a better way of doing this.

Reply:
@gabrielcsapo I think you should likely simplify this further, and decouple from the above helpers, and merger.

An example would look like

```js

// your deferred helper
function defer() {
  const result = {};
  result.promise = new Promise((resolve, reject) {
    result.resolve = resolve;
    result.reject =  reject;
  });
  return result;
}


class WaitingPlugin extends Plugin {
  constructor(inputNodes) {
      super(inputNodes);
      this._waiter = defer();
      this.buildCount = 0;
   }

   resolve(value) {
     this._waiter.resolve(value);
   }

   reject(reason) {
     this._waiter.resolve(reason);
   }

   async build() {
     try {
       await this._defer.promise;
     } finally {
       this.buildCount++;
       this._waiter = defer();
     }
   }
}

const first = new WaitingPlugin();
const second = new WaitingPlugin([first]);
const builder = new Builder(second);


```

   }

  
}
```

Reply:
^^ would not only address robs concern, but also make it very easy to test different combinations if needed.